src/UseCases/HandlePayPalPaymentNotification/HandlePayPalPaymentNotificationUseCase.php

```diff
@@ -131,4 +143,51 @@ private function newPayPalDataFromRequest( PayPalNotificationRequest $request ):
 			->setPaymentTimestamp( $request->getPaymentTimestamp() );
 	}
 
+	private function donationWasBookedWithDifferentTransactionId( Donation $donation,
+																  PayPalNotificationRequest $request ): bool {
+		/**
+		 * @var PayPalPayment $payment
+		 */
+		$payment = $donation->getPaymentMethod();
+
+		if ( !$donation->isBooked() ) {
+			return false;
+		}
+
+		if ( $request->getTransactionId() === $payment->getPayPalData()->getPaymentId() ) {
+			return false;
+		}
+
+		if ( $payment->getPayPalData()->hasChildPayment( $request->getTransactionId() ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	private function createChildDonation( Donation $donation, PayPalNotificationRequest $request ) {
+		$childPaymentMethod = new PayPalPayment( $this->newPayPalDataFromRequest( $request ) );
+		$payment = $donation->getPayment();
+		$childDonation = new Donation(
+			null,
+			Donation::STATUS_EXTERNAL_BOOKED,
+			$donation->getDonor(),
+			new DonationPayment( $payment->getAmount(), $payment->getIntervalInMonths(), $childPaymentMethod ),
```

Comment:
The old code [copies the values from the parent donation](https://github.com/wmde/fundraising/blob/master/spenden/paypal_handler.php#L227-L240), I just emulated that behavior. This is *not* for the recurring donations, where there will be  extra code for that, where interval is set to `0` and amount is taken from `$request`.